lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -1280,7 +1286,7 @@ function strictLookup(requireTerminal, compiler, parts, i, type) {
       '(',
       stack,
       ', ',
-      compiler.quotedString(parts[i]),
+      compiler.quotedString(parts[len]),
```

Comment:
Can you explain this change to me? Before #1855, this was always `parts[0]` (because `i` was `0`). The fix changed this to be the start-part-index. Now you changed it to be `parts.length`.

Shouldn't a test have failed somewhere?

Reply:
after the for-loop, after the `for (; i < len; i++) { ... }`, "i" always has the same value as "len", even before #1855.